lib/kitchen/driver/oci/blockstorage.rb

```diff
@@ -131,14 +131,25 @@ def attachment_response(attachment_id)
         end
 
         def volume_details(volume)
-          OCI::Core::Models::CreateVolumeDetails.new(
+          details = OCI::Core::Models::CreateVolumeDetails.new(
             compartment_id: oci.compartment,
             availability_domain: config[:availability_domain],
             display_name: volume[:name],
             size_in_gbs: volume[:size_in_gbs],
             vpus_per_gb: volume[:vpus_per_gb] || 10,
             defined_tags: config[:defined_tags]
           )
+          add_volume_source_details(details, volume)
+        end
+
+        def add_volume_source_details(details, volume)
+          if volume.key?(:source_details)
+            details.source_details = OCI::Core::Models::VolumeSourceFromVolumeDetails.new(
+              type: volume[:source_details][:type],
+              id: volume[:source_details][:id]
+            )
+          end
+          details
```

Comment:
I would like to see this split out a bit more elegantly.  there are some common attributes to the `OCI::Core::Models::CreateVolumeDetails` that both a new volume and a cloned volume need.  There are others like `size_in_gbs` and `vpus_per_gb` that do not appear to be necessary when cloning.

Reply:
`size_in_gbs` and `vpus_per_gb` are not needed to clone but they can be used together to create a clone with different sizes and vpus so long as the size is equal to or bigger than the original. 

I can make any suggested changes but `source_details` value is the only value in the map that gets added to the details to make it be a clone so I think we should still support those values in case someone wants to try that out in a test kitchen environment.

Reply:
I created separate clone methods now, we can tune them if we like but now they're separate so they can have different parameters if we like. 

```ruby
        def create_clone_volume(volume)
          clone_volume_name = clone_volume_display_name(volume[:volume_id])
          info("Creating <#{clone_volume_name}>...")
          result = api.blockstorage.create_volume(volume_clone_details(volume, clone_volume_name))
          response = volume_response(result.data.id)
          info("Finished creating <#{clone_volume_name}>.")
          [response, final_state(response)]
        end
        
        def volume_clone_details(volume, clone_volume_name)
          OCI::Core::Models::CreateVolumeDetails.new(
            compartment_id: oci.compartment,
            availability_domain: config[:availability_domain],
            display_name: clone_volume_name,
            defined_tags: config[:defined_tags],
            size_in_gbs: volume[:size_in_gbs],
            vpus_per_gb: volume[:vpus_per_gb],
            source_details: OCI::Core::Models::VolumeSourceFromVolumeDetails.new(id: volume[:volume_id])
          )
        end
```